src/components/marquee/index.tsx

```diff
@@ -0,0 +1,118 @@
+  const fixedDuration = isHorizontal ? duration || calcDuration() : duration || DEFAULT_DURATION;
+
+  const [viewLayout, setViewLayout] = useState<LayoutRectangle | undefined>(undefined);
+  const [textLayout, setTextLayout] = useState<LayoutRectangle | undefined>(undefined);
+
+  const offset = useSharedValue<number | undefined>(undefined);
+
+  let initialOffset = 0;
+  let axisX = false;
+  let axisY = false;
+
+  if (direction === MarqueeDirections.RIGHT || direction === MarqueeDirections.LEFT) {
+    axisX = true;
+  } else {
+    axisY = true;
+  }
+
+  const onLayoutView = (event: LayoutChangeEvent) => {
+    setViewLayout(event.nativeEvent.layout);
+  };
+
+  const onLayoutText = (event: LayoutChangeEvent) => {
+    setTextLayout(event.nativeEvent.layout);
+  };
+
+  const startAnimation = (fromValue: number, toValue: number, backToValue: number) => {
+    initialOffset = fromValue;
+    offset.value = initialOffset;
+
+    offset.value = withRepeat(withTiming(toValue, {duration: fixedDuration, easing: Easing.linear}),
+      numberOfReps,
+      false,
+      finished => {
+        if (finished) {
+          offset.value = initialOffset;
+          offset.value = withTiming(backToValue, {
+            duration: 2000
+          });
+        }
+      });
+  };
+
+  const translateStyle = useAnimatedStyle(() => {
+    if (offset.value) {
+      return {
+        transform: [{translateX: axisX ? offset.value : 0}, {translateY: axisY ? offset.value : 0}]
+      };
+    }
+    return {};
+  });
+
+  useEffect(() => {
+    if (viewLayout && textLayout) {
+      switch (direction) {
+        case MarqueeDirections.RIGHT:
+          startAnimation(-textLayout.width, viewLayout.width, 0);
+          break;
+        case MarqueeDirections.LEFT:
+          startAnimation(viewLayout?.width, -textLayout.width, viewLayout.width - textLayout.width);
+          break;
+        case MarqueeDirections.UP:
+          startAnimation(viewLayout.height, -textLayout.height, viewLayout.height - textLayout.height);
+          break;
+        case MarqueeDirections.DOWN:
+          startAnimation(-textLayout.height, viewLayout.height, 0);
+          break;
+      }
+    }
+  }, [viewLayout, textLayout]);
+
+  return (
+    <View style={[styles.container, containerStyle]} onLayout={onLayoutView}>
+      <View
+        reanimated
+        style={[
+          {position: 'absolute', width: !isHorizontal || textLayout?.width ? textLayout?.width : '400%'},
+          translateStyle
+        ]}
+      >
+        <Text style={[styles.text, labelStyle]} onLayout={onLayoutText}>
+          {label}
+        </Text>
+      </View>
+      <Text style={[styles.text, labelStyle, {color: 'transparent'}]} numberOfLines={1}>
+        {label}
+      </Text>
+    </View>
+  );
+}
+
+export {MarqueeProps, MarqueeDirections};
+
+export default asBaseComponent<MarqueeProps>(forwardRef(Marquee));
```

Comment:
No reason to wrap here with our forwardRef HOC, you don't use this as well in the component so it's not necessary. 
If you are not clear on how forwardRef works and when to use it, let's go over it